python/pyspark/worker.py

```diff
@@ -123,15 +120,9 @@ def wrapped(key_series, value_series):
                 "Number of columns of the returned pandas.DataFrame "
                 "doesn't match specified schema. "
                 "Expected: {} Actual: {}".format(len(return_type), len(result.columns)))
+        return result
 
-        # Assign result columns by schema name if user labeled with strings, else use position
-        if assign_cols_by_name and any(isinstance(name, basestring) for name in result.columns):
```

Comment:
Eh, @BryanCutler, sorry if I runshed to read but where did this logic go?

Reply:
Oops, okie. The logic was actually duplicated with `_create_batch`.

Reply:
yup, that's correct